### ISSUE TITLE
[App]: Images not showing on panels with more than one image

### ISSUE THREAD

**Issue by janpisl** [wandb/wandb#6534]
### Current Behavior

I log image to wandb. By default there is a panel created for every unique image name. This works fine and I can see all images that I've logged.

The issue arises when I want to visualize them in a single panel side-by-side. Suddenly I get a message as if there were no images logged under a given name.

But if I remove the other image names from the panel and leave only a single one, then it appears again.

### Expected Behavior

I want to be able to add more images in a single panel.

### Screenshots

No issue when only a single image is in the panel:

<img width="1795" alt="Screenshot 2023-11-01 at 15 37 02" src="https://github.com/wandb/wandb/assets/32265011/d50703b5-5202-4611-95f2-9b5c55a62916">

When adding another image, suddenly it appears as if the original image weren't there:

<img width="1795" alt="Screenshot 2023-11-01 at 15 37 10" src="https://github.com/wandb/wandb/assets/32265011/57dc23f7-0054-4b6c-9fef-f33efc06c922">

### Environment

OS: Ubuntu

Browsers: Chrome

Version:

**Comment by umakrishnaswamy** [comment]
Hey @janpisl - could you provide a sample code snippet for how you are logging these images?

**Comment by janpisl** [comment]
Sure, here is my entire pl.LightningModule + how I use it in a pl.Trainer. The logging happens in on_train_epoch_end()

`class CoffeeSegmenter(pl.LightningModule):
    def __init__(self, model, ignore_index=-1):
        super().__init__()

        
        self.model = model
        self.ignore_index = ignore_index

        self.IoU = BinaryJaccardIndex(ignore_index=self.ignore_index)
        
        #This is following the pl API. 
        self.images_to_log = {
            'image' : {
                'train' : None,
                'val' : None
            },
            'target' : {
                'train' : None,
                'val' : None
            },
            'prediction' : {
                'train' : None,
                'val' : None
            }
        }
        

    def training_step(self, batch, batch_idx):
        x, y = batch

        
        y_hat = self.model(x)
        loss = masked_BCE(y_hat, y, self.ignore_index)
        preds = torch.sigmoid(y_hat)
        
        self.log("train_loss", loss)
        wandb.log({"train_loss": loss})
        
        self.images_to_log['image']['train'] = get_normalized_image(x[0, 1:4].cpu().numpy().transpose(1,2,0))
        self.images_to_log['target']['train'] = y[0]
        self.images_to_log['prediction']['train'] = preds[0]
        

        log_dict = {"train_loss": loss}
        
        return {"loss": loss, "log": log_dict}

    
    def _shared_eval_step(self, batch, batch_idx, dataset_name):
        x, y = batch
        y_hat = self.model(x)
        
        loss = masked_BCE(y_hat, y, self.ignore_index)
        preds = torch.sigmoid(y_hat)

        
        idx_img_to_log = random.randint(0,y.shape[0]-1) #Randomly pick one sample from the batch 
        if len(np.unique(y[idx_img_to_log].cpu().numpy())) == 3: #Ensure it's positive because at the end of the batch i have a lot of negative examples and that's not informative
            self.images_to_log['image'][dataset_name] = get_normalized_image(x[idx_img_to_log, 1:4].cpu().numpy().transpose(1,2,0))
            self.images_to_log['target'][dataset_name] = y[idx_img_to_log]
            self.images_to_log['prediction'][dataset_name] = preds[idx_img_to_log]

        iou = self.IoU(preds, y.int())
        
        self.log(f"{dataset_name}_loss", loss)
        self.log(f"{dataset_name}_iou", iou)
        
        return loss, iou        

    def validation_step(self, batch, batch_idx):

        loss, iou = self._shared_eval_step(batch, batch_idx, 'val')
        
        log_dict = {'val_loss' : loss}
        return {"val_loss": loss, "log": log_dict, "val_iou": iou}

    
    def on_train_epoch_end(self):
      
        
        # After each epoch, log val and train triplets of image-target-prediction
        # This should be called after validation according to:
        #https://lightning.ai/docs/pytorch/stable/common/lightning_module.html#hooks
     
        for key, value in self.images_to_log['image'].items():
            wandb.log({
            f"{key}_image" : 
                wandb.Image(
                    value,
                    mode='RGB'
                    )}),
                
            self.images_to_log['image'][key] = None #Set to None to make sure i don't log the same again
        
        for _key in ['target', 'prediction']:   
            
            for key, value in self.images_to_log[_key].items():
                wandb.log({
                f"{key}_{_key}": 
                    wandb.Image(
                        value,
                        mode='L'
                        )}),  
                self.images_to_log[_key][key] = None
   


wandb_logger = WandbLogger(project="Deforestino_v2")
pl.seed_everything(420)
    

deeplab = smp.DeepLabV3Plus(
    #encoder_name="resnet34",        
    encoder_weights="imagenet",     
    in_channels=13,                  
    classes=1,                      
)    

coffee_segmenter = CoffeeSegmenter(deeplab)

datamodule = DeforestinoDataModule(
    config=config
    )

trainer = pl.Trainer(max_epochs=5, 
                     logger=wandb_logger, 
                     fast_dev_run=False,
                    #log_every_n_steps=5
                    )
trainer.fit(model=coffee_segmenter, datamodule=datamodule)
wandb.finish()

   
`

**Comment by umakrishnaswamy** [comment]
hey @janpisl - given the script you sent over, this behavior is expected. we only support viewing side-by-side images if you log these images at the same time (and in your script, you are doing so in different places). an example of 'logging these images at the same time' would be something like the following:

`wandb.log({"taco": wandb.Image('taco.jpeg'), 'tree': wandb.Image('tree.jpeg')})` 

here's a [run](https://wandb.ai/uma-wandb/image-side/runs/b27te6qt?workspace=user-uma-wandb) that shows the above example of side-by-side images

**Comment by umakrishnaswamy** [comment]
Hi @janpisl, since we have not heard back from you we are going to close this request. If you would like to re-open the conversation, please let us know!
